portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,63 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  ArrayList<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //Convert the ArrayList of messages as JSON
+    String json = convertToJson(comments);
+
+    //Send the JSON as response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //Get inputs from the form
+    String username = request.getParameter("username");
+    String comment = request.getParameter("comment");
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("username", username);
+    commentEntity.setProperty("comment", comment);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    String commentDetails = "Username: " + username + "; Comment: " + comment;
+    comments.add(commentDetails);
+
+    //Respond
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println(comments.toString());
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * Converts the randomList instance into a JSON string using the Gson library.
+   */
+  private String convertToJson(ArrayList<String> list) {
+    Gson gson = new Gson();
```

Comment:
You might want to create an instance variable for this in the class so it can be reused. Objects like this are often relatively expensive to build, so you want to avoid doing so on every method call.

Reply:
I ended up getting rid of this function and have this line at the beginning of the class; but I am not sure whether that's better or no? Is there any other way I could have done it?